susemanager-utils/susemanager-sls/src/states/uyuni_config.py

```diff
@@ -0,0 +1,759 @@
+                                if new_ma not in current_managed_channels})
+
+        managed_changes.update({old_ma: False for old_ma in (current_managed_channels or [])
+                                if old_ma not in new_managed_channels})
+
+        subscribe_changes: Dict[str, bool] = {}
+        for new_channel in (new_subscribe_channels or []):
+            if new_channel not in (current_subscribe_channels or []) or not managed_changes.get(new_channel, True):
+                subscribe_changes[new_channel] = True
+
+        for curr_channel in (current_subscribe_channels or []):
+            if not (curr_channel in new_subscribe_channels or curr_channel in new_managed_channels):
+                if not __salt__['uyuni.channel_software_is_globally_subscribable'](curr_channel,
+                                                                                   org_admin_user,
+                                                                                   org_admin_password):
+                    subscribe_changes[curr_channel] = False
+        changes = {}
+        if managed_changes:
+            changes['manageable_channels'] = managed_changes
+        if subscribe_changes:
+            changes['subscribable_channels'] = subscribe_changes
+        return changes
+
+    def manage(self, login: str, password: str,
+               manageable_channels: Optional[List[str]] = [],
+               subscribable_channels: Optional[List[str]] = [],
+               org_admin_user: str = None, org_admin_password: str = None) -> Dict[str, Any]:
+        """
+        User channels management present implementation
+
+        :param login: user login ID
+        :param password: user password
+        :param manageable_channels: channels user can manage
+        :param subscribable_channels: channels user can subscribe
+        :param org_admin_user: organization administrator username
+        :param org_admin_password: organization administrator password
+        :return: dict for Salt communication
+        """
+        try:
+            current_roles = __salt__['uyuni.user_list_roles'](login, password=password)
+            current_manageable_channels = __salt__['uyuni.channel_list_manageable_channels'](login, password)
+            current_subscribe_channels = __salt__['uyuni.channel_list_my_channels'](login, password)
+        except Exception as exc:
+            return StateResult.state_error(login,
+                                           comment="Error managing user channels '{}': {}".format(login, exc))
+            pass
+
+        if "org_admin" in current_roles or "channel_admin" in current_roles:
+            return StateResult.state_error(login, "Channels access cannot be managed, "
+                                                "User can manage all channel in the organization "
+                                                "(\"org_admin\" or \"channel_admin\" role).")
```

Comment:
```suggestion
        if "org_admin" in current_roles or "channel_admin" in current_roles:
            return StateResult.state_error(login, "Channels access cannot be changed, because"
                                                "the target user can manage all channels in the organization "
                                                "(having an \"org_admin\" or \"channel_admin\" role).")
```